### PR TITLE
GEOMETRY_64_48 and drawStringf method added

### PR DESCRIPTION
**Pull request by thorsten-l**
I did only to few little changes to get my 64x48 Shield working and i really needed the possibility to do a 'printf' like formatted output. So i implemented a drawStringf method.

### CODE REVIEW

**Comment by Testato** [comment]
This PR solve the Issue https://github.com/ThingPulse/esp8266-oled-ssd1306/issues/73#issuecomment-493634672
Please accept it

**Comment by marcelstoer** [comment]
@thorsten-l sorry the PRs weren't processed in due time over the last year. Can I ask you to resolve the conflicts in src/OLEDDisplay.cpp? I had to merge a large PR from @helmut64 first, sorry.

File: src/OLEDDisplay.cpp [hunk 1]
```diff
@@ -801,15 +810,20 @@ int OLEDDisplay::_putc(int c) {
...
   	case GEOMETRY_128_64:
     	this->displayWidth = 128;
     	this->displayHeight = 64;
 		break;
-	case GEOMETRY_128_32:
+	  case GEOMETRY_128_32:
     	this->displayWidth = 128;
     	this->displayHeight = 32;
 		break;
+    case GEOMETRY_64_48:
```
Comment (marcelstoer) [inline src/OLEDDisplay.cpp]: Can you please fix the formatting in this `switch` statement (indentation), thanks.

Comment (LennartHennigs) [reply src/OLEDDisplay.cpp:823]: could you please approve this change request? it has been opened since 2018 and it basically works...

Comment (marcelstoer) [reply src/OLEDDisplay.cpp:823]: I approved it a year ago and we would have merged it if all feedback had been addressed or if we had been given permission to edit the file in question ourselves.

**Comment by LennartHennigs** [comment]
please reopen this PR and accept the fix

File: src/OLEDDisplay.cpp (hunk 1 above)
Comment (LennartHennigs) [reply src/OLEDDisplay.cpp:823]: Hey Marcel, 
thanks for replying. 
Isn't the only todo open: to fix the indenting of one code block?

Cheers
l.

**Comment by rin67630** [comment]
MiniOLED is prefectly working , but DrawStringf is only documented, does not compile...
Can we hope for a Christmas gift?

**Comment by thorsten-l** [comment]
`
#include <Arduino.h>
#include <SSD1306Brzo.h>

SSD1306Brzo display(0x3c, D2, D1, GEOMETRY_64_48 );

static char buffer[32];
static unsigned long counter;

void setup() {
  Serial.begin(115200);
  counter = 0l;
  display.init();
  display.flipScreenVertically();
}

void loop() {
  Serial.printf( "c=%lu\n", counter );
  display.clear();
  display.drawStringf( 10, 10, buffer, "c=%lu", counter );
                            // ^^^^^^ shared buffer to save memory
  display.display();
  delay( 1000 );
  counter++;
}
`
I just compiled and run this sample code successfully, with PlatformIO.

**Comment by thorsten-l** [comment]
```
[env:d1_mini]
platform = espressif8266
board = d1_mini
framework = arduino

lib_deps = 
  Brzo I2C@1.3.3
  https://github.com/ThingPulse/esp8266-oled-ssd1306

monitor_speed = 115200
upload_speed = 230400
```
with this platformio.ini file

**Comment by marcelstoer** [comment]
Thank you Thorsten 👍 🥇

**Comment by thorsten-l** [comment]
[ESP-SSD1306-DrawStringf-Sample](https://github.com/thorsten-l/ESP-SSD1306-DrawStringf-Sample)
A tiny bit more complex sample for drawStringf. Successfully tested with ESP8266 (D1 mini) and ESP32 (LOLIN32).
To give you a complete sample project.

**Comment by marcelstoer** [comment]
Way cool Thorsten! Thanks again for offering such excellent user support.

**Pull request merged**